### PR TITLE
Make code more easy to read

### PR DESCRIPTION
**Pull request by kevinanew**
Describe what this patch does to fix the issue.
When I read the code, I feel something is hard to understand.
So I change it.

### CODE REVIEW

File: src/flask/app.py [hunk 1]
```diff
@@ -1256,11 +1260,11 @@ def index():
             )
 
         if provide_automatic_options is None:
-            if "OPTIONS" not in methods:
+            if "OPTIONS" in methods:
+                provide_automatic_options = False
+            else:
```
Comment (sukurcf) [inline src/flask/app.py]: why not 
`provide_automatic_options = False if "OPTIONS" in methods else True`

Comment (davidism) [reply src/flask/app.py]: Because there's other stuff going on in the `else` block.

Comment (ThiefMaster) [reply src/flask/app.py]: Absolutely not. First of all, this is just adding more complexity than needed as your code is just a more convoluted way of this:

    provide_automatic_options = "OPTIONS" not in methods

But then, if you look a the whole code, you'll notice that the code in the else branch does more than just setting the variable to True...

File: src/flask/app.py [hunk 2]
```diff
@@ -2146,11 +2150,11 @@ def create_url_adapter(self, request):
...
-            subdomain = (
-                (self.url_map.default_subdomain or None)
-                if not self.subdomain_matching
-                else None
-            )
+            if self.subdomain_matching:
+                subdomain = None
+            else:
+                subdomain = self.url_map.default_subdomain or None
+
```
Comment (sukurcf) [inline src/flask/app.py]: Why not
`subdomain = None if self.subdomain_matching else self.url_map.default_subdomain`

Comment (ThiefMaster) [reply src/flask/app.py]: That would not convert `''` to `None` as the current code does.

Comment (kevinanew) [reply src/flask/app.py]: @sukurcf Because it's will too long (more than 78 characters), that's why.

**Comment by davidism** [comment]
I like where you've expanded the code into full `if/else` statements. However, I'm going to back out the wrapping changes, as they don't really match what this PR is about and aren't really consistent.

**Pull request merged**